WordPress/gutenberg
# Fix: Table block doesn't keep background color

## What?
In table block, the background color is not reflected correctly in the editor if **only the background color is selected from the color palette**.

## Why?
If only the background color is selected in the color palette, html is generated in which only the class is given to the table tag and no style attribute is given, as shown below:

```html

<figure class="wp-block-table">
  <table class="has-pale-pink-background-color has-background">
    <tbody>
      <tr><td>aaaa</td><td>bbbb</td></tr>
      <tr><td>cccc</td><td>dddd</td></tr>
    </tbody>
  </table>
</figure>

```

Therefore, the backward-compatibility support for initial table blocks with colors such as `subtle-light-gray` also affects such HTML:

```javascript
isEligible: ( attributes ) => {
	// Return true even if only the background color is selected from the color palette
	return attributes.backgroundColor && ! attributes.style;
},
```

## How?
In this PR, the initial table block migrate is now limited to only those cases where the background color slug is included in oldColors ( `subtle-light-gray` / `subtle-pale-green` / `subtle-pale-blue` / `subtle-pale-pink` )

## Testing Instructions

- Insert table block.
- Select any background color from the color palette.
- Save post and verify that the background color is kept after reloading the browser.

## Screenshots or screencast 
before this PR ( on trunk ): 

https://user-images.githubusercontent.com/54422211/158299417-944fc525-53ab-41a6-a311-4ad62da2bef3.mp4

## About edge case

If the same slug is defined as the old color variation of the table in the custom color palette, it will be considered the initial table block and the oldColors color value will take precedence.

https://user-images.githubusercontent.com/54422211/158299383-fd27aed4-67b2-40ed-825e-499acd2e2576.mp4

If there is any way to avoid these edge cases, please advise.
